hack/dci-merge.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# script called from a merge queue branch of the redhatci/ansible-collection-redhatci-ocp repo
+
+VIRT=
+DCI_QUEUE=
+SNO_DCI_QUEUE=
+SUPPORTED_HINTS=
+NO_DCI_QUEUE=${NO_DCI_QUEUE:-}
+NO_COMMENT=
+UPGRADE=
+UPGRADE_ARGS=
+APP_NAME=
+APP_ARGS=
+FORCE_CHECK=
+
+# shellcheck disable=SC1091
+if [ -r /etc/dci-openshift-agent/config ]; then
+    . /etc/dci-openshift-agent/config
+fi
+
+GH_HEADERS=(
+    "Accept: application/vnd.github.v3+json"
+    "Authorization: token ${GITHUB_TOKEN}"
+)
+
+# Lookup the merge commits and get their PR descriptions to detect Test-Hints: strings
+COMMIT=HEAD
+VIRT=
+while true; do
+    PR=$(git log -1 "$COMMIT"|grep -oP 'Merge pull request #\K\d+')
```

Comment:
What about using: 
```suggestion
    PR=$(git log --merges "$COMMIT"|grep -m1 -oP 'Merge pull request #\K\d+')
```

Since we only care about merges

Reply:
yes but I only want the PRs that are part of this merge group so I want to stop when not finding a merge commit.